pkg/apis/storage/validation/validation.go

```diff
@@ -459,3 +471,56 @@ func validateVolumeLifecycleModes(modes []storage.VolumeLifecycleMode, fldPath *
 
 	return allErrs
 }
+
+// ValidateStorageCapacityName checks that a name is appropriate for a
+// CSIStorageCapacity object.
+var ValidateStorageCapacityName = apimachineryvalidation.NameIsDNSSubdomain
+
+// ValidateCSIStorageCapacity validates a CSIStorageCapacity.
+func ValidateCSIStorageCapacity(capacity *storage.CSIStorageCapacity) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&capacity.ObjectMeta, false, ValidateStorageCapacityName, field.NewPath("metadata"))
+	allErrs = append(allErrs, apivalidation.ValidateCSIDriverName(capacity.DriverName, field.NewPath("driverName"))...)
+	if capacity.NodeTopology != nil {
+		nodeTopoFldPath := field.NewPath("nodeTopology")
+		allErrs = append(allErrs, apivalidation.ValidateNodeSelector(capacity.NodeTopology, nodeTopoFldPath)...)
+		allErrs = append(allErrs, validateNoFieldSelector(capacity.NodeTopology, nodeTopoFldPath)...)
+	}
+	for _, msg := range apivalidation.ValidateClassName(capacity.StorageClassName, false) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("storageClassName"), capacity.StorageClassName, msg))
+	}
+	// "capacity" doesn't need to be validated, any value is allowed.
```

Comment:
We do validate the capacity field in other objects like PVC/PV: https://github.com/kubernetes/kubernetes/blob/3748b85d3541ff91d6d9653942e2eae050a4b5aa/pkg/apis/core/validation/validation.go#L1599